doc/reference/pipeline_spec.md

```diff
@@ -619,6 +617,24 @@ The root mount point is at `/pfs`, which contains:
   name if not specified.
 - `/pfs/out` which is where you write any output.
 
-### Output Formats
-
-PFS supports data to be delimited by line, JSON, or binary blobs.
+# Environment Variables
+
+There are several environment variables that get injected into the user code
+before it runs. They are:
+
+- `PACH_JOB_ID` the id the currently run job.
+- `PACH_OUTPUT_COMMIT_ID` the id of the commit being outputted to.
+- For each input there will be an envirnment variable with the same name
+    defined to the path of the file for that input. For example if you are
+    accessing an input called foo from the path `/pfs/foo` which contains a file
+    called `bar` then the environment variable `foo` will have the value
+    `/pfs/foo/bar`. 
+
+In addition to these environment variables Kubernetes also injects others for
+Services that are running inside the cluster. These allow you to connect to
+outside those services, which can be powerful but also can be hard to reason
+about, as processing me be retried multiple times. Interaction with outside
```

Comment:
typo: "processing might be retried"?

Reply:
How does processing being retried make it hard to reason about? Perhaps explain with simple example?